security.go

```diff
@@ -0,0 +1,186 @@
+package goa
+
+import (
+	"errors"
+	"net/http"
+
+	"golang.org/x/net/context"
+)
+
+// SecurityMethod extracts the security method (OAuth2Security,
+// BasicAuthSecurity, APIKeySecurity or OtherSecurity) from the
+// request context.
+//
+// This is to be used from within a security middleware
+// implementation. If you do get into your middleware code, it is
+// guaranteed that the request will hold a valid SecurityMethod.  You
+// can then typecast it and validate the proper credentials.
+func SecurityMethod(ctx context.Context) interface{} {
+	return ctx.Value(securityMethodKey)
+}
+
+// Scopes extracts from a request the scopes relevant to the action
+// being executed.
+//
+// Call this from within your Security middleware implementation.
+//
+// Scopes can be empty
+func Scopes(ctx context.Context) (out []string) {
+	scopes, ok := ctx.Value(securityScopesKey).([]string)
+	if !ok {
+		return
+	}
+	return scopes
+}
+
+// securityMiddleware represents a security method middleware, used
+// for deduplicating the different security method's methods!
+type securityMiddleware struct {
+	middleware Middleware
+	method     interface{}
+
+	// MethodName is the name specified for this authentication method
+	// which will appear in the Swagger file.
```

Comment:
Reference to swagger again, consider this in the absolute instead - in this case do we need te MethodName here? We need it in the design definitions - not sure we need it at runtime?